### ISSUE TITLE
Dart rules in flutter/engine should use package_config.json instead of .packages

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#60999]
The .packages file doesn't contain language version information, which can block rolls of null safe packages through the engine. It should be straightforward to switch to `.dart_tool/package_config.json`, and this is already done for the flutter_frontend_server and the const_finder libraries.

The remaining dart rules should be audited and updated.

**Comment by jonahwilliams** [comment]
fyi @zanderso

**Comment by zanderso** [comment]
@jonahwilliams Looks like you're on this already, but could you link to an example location that needs an update? Thanks!

**Comment by jonahwilliams** [comment]
Here is an example commit that fixed const_finder:

https://github.com/flutter/engine/commit/35b5aa5af24925eec3f3bf17543a8662831f57b3

Notable differences: dot_packages in the gn template updated to point at .dart_tool/package_config and the generate package_config script updated to generate this on gclient sync (so that the language versions are up to date for rolls). Keeping a checked in .packages isn't sufficient anymore

**Comment by jonahwilliams** [comment]
Blocked by https://github.com/dart-lang/sdk/issues/42614 for the kernel_compiler target

**Comment by jonahwilliams** [comment]
Status: currently waiting on a dart roll
